### PR TITLE
improvement: Fix types in form item

### PR DESCRIPTION
**Pull request by fa93hws**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/form/FormItem.tsx [hunk 1]
```diff
@@ -24,22 +24,17 @@ type ChildrenType = RenderChildren | React.ReactNode;
...
+  ({ children }: MemoInputProps) => <>{children}</>,
   (prev, next) => {
     return prev.value === next.value && prev.update === next.update;
   },
 );
 
-export interface FormItemProps
-  extends FormItemLabelProps,
-    FormItemInputProps,
-    Omit<RcFieldProps, 'children'> {
```
Comment (fa93hws) [inline components/form/FormItem.tsx:42]: `children` has already been omitted at https://github.com/ant-design/ant-design/pull/22962/files#diff-94b7635a3702d37a73e4971da7dda90bR21

File: components/form/FormItem.tsx [hunk 2]
```diff
@@ -24,22 +24,17 @@ type ChildrenType = RenderChildren | React.ReactNode;
...
-  children: any;
+  children: React.ReactNode;
 }
 
-const MemoInput = React.memo<MemoInputProps>(
-  ({ children }) => {
-    return children;
-  },
+const MemoInput = React.memo(
+  ({ children }: MemoInputProps) => <>{children}</>,
```
Comment (zombieJ) [inline components/form/FormItem.tsx:31]: Why create a fragment to wrap this?

Comment (afc163) [reply components/form/FormItem.tsx:31]: @fa93hws ping

Comment (fa93hws) [reply components/form/FormItem.tsx:31]: It can be `as JSX.Element` to pass the type check if the fragment is not preferred.

**Pull request merged**